cov.py

```diff
@@ -31,51 +31,65 @@
-                opts, timeout, simulator, simulator_yaml, custom_target,
-                    target, stop_on_first_error):
+def collect_cov(log_dir, out, core, iss, testlist, batch_size, lsf_cmd, steps, \
+                opts, timeout, simulator, simulator_yaml, custom_target, \
+                target, stop_on_first_error):
   """Collect functional coverage from the instruction trace
 
   Args:
-    log_dir          : ISS log directory
-    out              : Output directory
-    iss              : Instruction set simulator
-    testlist        : Testlist of the coverage test
-    batch_size       : Number of trace CSV to process per test
-    lsf_cmd          : LSF command used to run the instruction generator
-    steps            : csv:log to CSV, cov:sample coverage
-    opts             : Additional options to the instruction generator
-    timeout          : Timeout limit in seconds
-    simulator        : RTL simulator used to run
-    simulator_yaml   : RTL simulator configuration file in YAML format
-    custom_target    : Path for the custom target dir
-    target           : Predefined target
+    log_dir             : Trace log directory
+    out                 : Output directory
+    core                : RISC-V core to be used for analysis
+    iss                 : Instruction set simulator
+    testlist            : Testlist of the coverage test
+    batch_size          : Number of trace CSV to process per test
+    lsf_cmd             : LSF command used to run the instruction generator
+    steps               : csv:log to CSV, cov:sample coverage
+    opts                : Additional options to the instruction generator
+    timeout             : Timeout limit in seconds
+    simulator           : RTL simulator used to run
+    simulator_yaml      : RTL simulator configuration file in YAML format
+    custom_target       : Path for the custom target dir
+    target              : Predefined target
     stop_on_first_error : will end run on first error detected
   """
   cwd = os.path.dirname(os.path.realpath(__file__))
   log_list = []
   csv_list = []
-  trace_log = ("%s/%s_trace_log" % (out, iss))
   logging.info("Processing trace log under %s" % log_dir)
-  run_cmd("find %s -name \"*.log\" | sort > %s" % (log_dir, trace_log))
-  with open(trace_log) as f:
-    for line in f:
-      line = line.rstrip()
-      log_list.append(line)
-      csv = line[0:-4] + ".csv"
-      csv_list.append(csv)
+  if core:
+    """
+    If functional coverage is being collected from an RTL core implementation,
+    the flow assumes that the core's trace logs have already been converted to
+    CSV files by the post_compare step of the flow.
+    """
+    trace_log = ("%s/%s_trace_log" % (out, core))
+    run_cmd("find %s -name \"*.csv\" | sort > %s" % (log_dir, trace_log))
+  else:
+    trace_log = ("%s/%s_trace_log" % (out, iss))
+    run_cmd("find %s -name \"*.log\" | sort > %s" % (log_dir, trace_log))
+    with open(trace_log) as f:
```

Comment:
I believe you need to remove the indent here?

Reply:
Yes, I've removed it locally. I found a bug with how I'm using assign_operand(), so I haven't updated this PR yet